### PR TITLE
Feat/create scoreboard

### PR DESCRIPTION
**Pull request by naikibro**
## Updates
- created the Scoreboard
- Map now uses Tiles
- added event listenner on Tile click ( TO IMPLEMENT )
- added color handling on scoreboard for future features

## future updates:
- pass the player to the scoreboard
- display the name of player inside scoreboard

REF: #16

### CODE REVIEW

**Comment by naikibro** [comment]
[Screencast from 04-29-2024 10:50:23 AM.webm](https://github.com/naikibro/BatailleNavale/assets/64978347/f1385c63-61ec-44d0-b838-d28f80bfa38d)

File: gui/src/map/Tile.java [hunk 1]
```diff
@@ -3,6 +3,8 @@
 public class Tile {
 
     private int id, x, y;
```
Comment (tteuru09) [inline gui/src/map/Tile.java]: On a plus besoin de l'attribut "id"

File: gui/src/map/Tile.java [hunk 2]
```diff
@@ -23,6 +25,37 @@ public class Tile {
...
-    // ----- S E T T E R S -----
+    // ----- G E T T E R S - S E T T E R S -----
+    public int getId() {
+        return id;
+    }
+
+    public void setId(int id) {
+        this.id = id;
+    }
+
```
Comment (tteuru09) [inline gui/src/map/Tile.java]: Du coup getter et setter ils sautent aussi

**Review by tteuru09 — COMMENTED** [review]
All Good, il faut juste hoaia enlever le "id" dans Tile, ses getters et setters hoaia aussi. 🤙🏽

File: gui/src/map/Tile.java (hunk 1 above)
Comment (naikibro) [reply gui/src/map/Tile.java]: should be good now

File: gui/src/map/Tile.java (hunk 2 above)
Comment (naikibro) [reply gui/src/map/Tile.java]: should be good now

**Pull request merged**